scripts/gtcadis.py

```diff
@@ -0,0 +1,205 @@
+                       'bi209']
+
+    names_dict = {nucname.id(x): y for x, y in zip(names_formatted, names)}
+
+    return names_dict
+
+
+def _cards(source):
+    cards = {"block1": {"isn": 16,
+                        "maxscm": '3E8',
+                        "maxlcm": '6E8',
+                        },
+             "block3": {"lib": "xsf21",  # name of cross section library
+                        "lng": 175,
+                        "maxord": 5,
+                        "ihm": 227,
+                        "iht": 10,
+                        "ihs": 11,
+                        "ifido": 1,
+                        "ititl": 1,
+                        "i2lp1": 0,
+                        "savbxs": 0,
+                        "kwikrd": 0
+                        },
+             "block5": {"source": source,
+                        "ith": 1,
+                        "isct": 5}
+             }
+    return cards
+
+
+def step1():
+    # Parse config file
+    config = ConfigParser.ConfigParser()
+    config.read(config_filename)
+
+    # Get user-input from config file
+    geom = config.get('step1', 'geom_file')
+    cells = [config.getint('step1', 'src_cell')]
+    src_vol = [config.getfloat('step1', 'src_vol')]
+    xmesh = config.get('step1', 'xmesh').split(',')
+    ymesh = config.get('step1', 'ymesh').split(',')
+    zmesh = config.get('step1', 'zmesh').split(',')
+
+    # Create structured mesh
+    sc = [np.linspace(float(xmesh[0]), float(xmesh[1]), float(xmesh[2]) + 1),
+          np.linspace(float(ymesh[0]), float(ymesh[1]), float(ymesh[2]) + 1),
+          np.linspace(float(zmesh[0]), float(zmesh[1]), float(zmesh[2]) + 1)]
+    m = Mesh(structured=True, structured_coords=sc)
+    m.mesh.save("blank_mesh.h5m")
+
+    # Generate 42 photon energy bins [eV]
+    #  First bin has been replaced with 1 for log interpolation
+    photon_bins = [1, 1e4, 2e4, 3e4, 4.5e4, 6e4, 7e4, 7.5e4, 1e5, 1.5e5, 2e5, 3e5, 4e5,
+                   4.5e5, 5.1e5, 5.12e5, 6e5, 7e5, 8e5, 1e6, 1.33e6, 1.34e6, 1.5e6, 1.66e6, 2e6,
+                   2.5e6, 3e6, 3.5e6, 4e6, 4.5e6, 5e6, 5.5e6, 6e6, 6.5e6, 7e6, 7.5e6, 8e6, 1e7,
+                   1.2e7, 1.4e7, 2e7, 3e7, 5e7]
+    # Convert to MEV
+    photon_bins = np.array([x / 1E6 for x in photon_bins])
+    # ICRP 74 flux-to-dose conversion factors
+    de = np.array([0.01, 0.015, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.1, 0.15, 0.2, 0.3,
+                   0.4, 0.5, 0.6, 0.8, 1, 2, 4, 6, 8, 10])
+    df = np.array([0.0485, 0.1254, 0.205, 0.2999, 0.3381, 0.3572, 0.378, 0.4066, 0.4399, 0.5172,
+                   0.7523, 1.0041, 1.5083, 1.9958, 2.4657, 2.9082, 3.7269, 4.4834, 7.4896,
+                   12.0153, 15.9873, 19.9191, 23.76])
+    # Convert to Sv/s per photon FLUX (not fluence)
+    df = np.array([x * 1E-12 for x in df])
+    # Convert pointwise data to group data for log interpolation
+    photon_spectrum = pointwise_collapse(
```

Comment:
variable naming?? this is really a response function since this is all about adjoints